provider/image.go

```diff
@@ -301,10 +286,45 @@ func getCachedImages(img Image, b resource.PropertyValue) []string {
 func setRegistry(r resource.PropertyValue) Registry {
 	var reg Registry
 	if !r.IsNull() {
-		reg.Server = r.ObjectValue()["server"].StringValue()
-		reg.Username = r.ObjectValue()["username"].StringValue()
-		reg.Password = r.ObjectValue()["password"].StringValue()
+		if !r.ObjectValue()["server"].IsNull() {
+			reg.Server = r.ObjectValue()["server"].StringValue()
+		}
+		if !r.ObjectValue()["username"].IsNull() {
+			reg.Username = r.ObjectValue()["username"].StringValue()
+		}
+		if !r.ObjectValue()["password"].IsNull() {
+			reg.Password = r.ObjectValue()["password"].StringValue()
+		}
 		return reg
 	}
 	return reg
 }
+
+func setArgs(a resource.PropertyValue) map[string]*string {
```

Comment:
Curious why these are called setX? They're pure functions that are doing some marshaling? The set prefix makes me expect they set something.

Reply:
replacing with `marshalX` - thank you that is a good point!